js/header-scroll.js

```diff
@@ -0,0 +1,13 @@
+let lastScrollY = window.scrollY;
+
+window.addEventListener('scroll', () => {
+  const header = document.querySelector('.md-header__inner');
+  if (window.scrollY > lastScrollY) {
+    // User is scrolling down - hide the header
+    header.classList.add('hidden');
+  } else {
+    // User is scrolling up - show the header
+    header.classList.remove('hidden');
+  }
+  lastScrollY = window.scrollY;
+});
```

Comment:
We can use the `.toggle` functionality for these scenarios, I think it would look cleaner and more readable :) 

```suggestion
let lastScrollY = window.scrollY;
const header = document.querySelector('.md-header__inner');

window.addEventListener('scroll', () => {
  // Toggle header visibility based on scroll direction
  header.classList.toggle('hidden', window.scrollY > lastScrollY);
  lastScrollY = window.scrollY;
});
```